src/core/cgroup.c

```diff
@@ -4062,38 +4066,49 @@ static int unit_get_memory_peak_raw(Unit *u, uint64_t *ret) {
         if (!r)
                 return -ENODATA;
 
-        return cg_get_attribute_as_uint64("memory", u->cgroup_path, "memory.peak", ret);
+        return cg_get_attribute_as_uint64("memory", u->cgroup_path, mem_attribute, ret);
 }
 
-int unit_get_memory_peak(Unit *u, uint64_t *ret) {
+static int unit_get_memory_attr_cached(Unit *u,  const char* mem_attribute, uint64_t* last, uint64_t *ret) {
         uint64_t bytes;
         int r;
 
         assert(u);
+        assert(mem_attribute);
+        assert(last);
         assert(ret);
 
-        if (!UNIT_CGROUP_BOOL(u, memory_accounting))
```

Comment:
Please don't drop this, else you can't properly distinguish the case when memory accounting is disabled from "real" -ENODATA.

It's OK to have redundant checks here I think.